core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -1179,4 +1238,61 @@ public void setNext(int next) {
             cachedNext = next;
         }
     }
+
+    public static class ShareFetchPartitionData {
+        private final FetchParams fetchParams;
+        private final String groupId;
+        private final String memberId;
+        private final Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData;
+        private final CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future;
+
+        public ShareFetchPartitionData(FetchParams fetchParams, String groupId, String memberId,
+                                       Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData,
+                                       CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future) {
+            this.fetchParams = fetchParams;
+            this.groupId = groupId;
+            this.memberId = memberId;
+            this.topicPartitionData = topicPartitionData;
+            this.future = future;
+        }
+
+        public FetchParams fetchParams() {
+            return fetchParams;
+        }
+
+        public String groupId() {
+            return groupId;
+        }
+
+        public String memberId() {
+            return memberId;
+        }
+
+        public Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData() {
+            return topicPartitionData;
+        }
+
+        public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future() {
+            return future;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(fetchParams, groupId, memberId, topicPartitionData, future);
+        }
+
+        @Override
+        public boolean equals(final Object obj) {
+            if (this == obj)
+                return true;
+            else if (obj == null || getClass() != obj.getClass())
+                return false;
+            else {
+                ShareFetchPartitionData that = (ShareFetchPartitionData) obj;
+                return groupId.equals(that.groupId) && memberId.equals(that.memberId) && Objects.equals(fetchParams,
+                        that.fetchParams) && Objects.equals(topicPartitionData, that.topicPartitionData) &&
+                        Objects.equals(future, that.future);
+            }
+        }
```

Comment:
Do we require it now? Seems not a map now.

Reply:
you're right. We don't need it since we don't need a comparison of objects.